File: apps/android/app/src/main/java/com/clipboardsync/StatusViewModel.kt
package com.clipboardsync

enum class SyncConnectionState {
    CONNECTED,
    DEGRADED,
    DISCONNECTED
}

data class StatusViewModel(
    val connectionState: SyncConnectionState,
    val syncedOutCount: Int,
    val syncedInCount: Int,
    val rejectedEventCount: Int,
    val trustedDeviceCount: Int,
    val pendingPairingCount: Int,
    val lastErrorMessage: String?
)

File: apps/windows/src/StatusViewModel.cs
namespace ClipboardSync.Windows;

public enum SyncConnectionState
{
    Connected,
    Degraded,
    Disconnected
}

public sealed class StatusViewModel
{
    public SyncConnectionState ConnectionState { get; set; } = SyncConnectionState.Disconnected;
    public int SyncedOutCount { get; set; }
    public int SyncedInCount { get; set; }
    public int RejectedEventCount { get; set; }
    public int TrustedDeviceCount { get; set; }
    public int PendingPairingCount { get; set; }
    public string LastErrorMessage { get; set; } = string.Empty;
}
